data/trie/patriciaMerkleTrie.go

```diff
@@ -603,12 +604,15 @@ func (tr *patriciaMerkleTrie) GetAllLeaves() (map[string][]byte, error) {
 }
 
 // GetAllLeavesOnChannel adds all the trie leaves to the given channel
-func (tr *patriciaMerkleTrie) GetAllLeavesOnChannel(leavesChannel chan *data.TrieLeaf) error {
-	tr.mutOperation.RLock()
+func (tr *patriciaMerkleTrie) GetAllLeavesOnChannel() chan core.KeyValueHolder {
+	//TODO pass a context for cancellation purposes when needed
+	leavesChannel := make(chan core.KeyValueHolder)
 
+	tr.mutOperation.RLock()
 	if tr.root == nil {
-		close(leavesChannel)
 		tr.mutOperation.RUnlock()
+		close(leavesChannel)
+
 		return nil
```

Comment:
return ch in order to not output a nil chan and break down stream code

Reply:
Done.